# Profitable App Profiles for the App Store and Google Play Markets

This project is for finding out the profitability of Apps on the following stores: App Store(Apple) and Google Play(Android).

I want to use this project to find out certain things about app on the different markets like which:
- Which apps are successful in which market?
- Which market has the most apps created (paid vs free)?
- Is the top games popular on Apple Store or Google Play (Clash of Clans, Afk Arena, etc)?

In [1]:
open_file = open('googleplaystore.csv')
from csv import reader
read_file = reader(open_file)
google_data = list(read_file)
google_header = google_data[0]
google_data = google_data[1:]

open_file = open('AppleStore.csv')
read_file = reader(open_file)
apple_data = list(read_file)
apple_header = apple_data[0]
apple_data = apple_data[1:]

## Opening and Exploring the Data

For this example we are going to be looking at two different data sets:

- [A data set](https://www.kaggle.com/lava18/google-play-store-apps) containing data about approximately 10,000 Android apps from Google Play, the data was collected in August 2018.
- [A data set](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps) containing data about approximately 7000 iOS apps from the App Store; the data was colelcted in July 2017.

The following function **explore_data(dataset, start, end, rows_and_columns)** has the following:

- dataset: this ask for the .csv in question will be looked through (this must be a list)
- start: the starting point where we will be looking at the data
- end: the end point where we will stop looking
- rows_and_columns: if true this will display the number of rows and columns in the dataset


In [2]:
def  explore_data(dataset, start, end, rows_and_columns= False):
    dataset_slice = dataset[start:end]
    for row in dataset_slice:
        print(row)
        print('\n') # adds a new (empty) line after each row
        
    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))

With our new function we can now look at the first 5 entires of each data set:

In [3]:
print(apple_header)
print('\n')
explore_data(apple_data, 0, 5, True)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1']


['284035177', 'Pandora - Music & Radio', '130242560', 'USD', '0.0', '1126879', '3594', '4.0', '4.5', '8.4.1', '12+', 'Music', '37', '4', '1', '1']


Number of rows: 7197
Number of columns: 16


In [4]:
print(google_header)
print('\n')
explore_data(google_data, 0, 5, True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Eve

## Deleting Wrong Data

In this section we will be deleting data that is wrong:

This is based on the disscussion post: [discussion post](https://www.kaggle.com/lava18/google-play-store-apps/discussion/66015)

In [5]:
print(len(google_data))
del google_data[10472]
print(len(google_data))

10841
10840


## Removing Duplicate Entries

With this we will look at removing entires that are duplicates. If we explored the entires we can see for example that 'Quick PDF Scanner + OCR FREE'
has some duplicates.

In [6]:
for app in google_data:
    name = app[0]
    if name == 'Quick PDF Scanner + OCR FREE':
        print(app)
print('\n')

['Quick PDF Scanner + OCR FREE', 'BUSINESS', '4.2', '80805', 'Varies with device', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'February 26, 2018', 'Varies with device', '4.0.3 and up']
['Quick PDF Scanner + OCR FREE', 'BUSINESS', '4.2', '80805', 'Varies with device', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'February 26, 2018', 'Varies with device', '4.0.3 and up']
['Quick PDF Scanner + OCR FREE', 'BUSINESS', '4.2', '80804', 'Varies with device', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'February 26, 2018', 'Varies with device', '4.0.3 and up']




In [7]:
duplicated_apps = []
unique_apps = []

for app in google_data:
    name = app[0]
    if name in unique_apps:
        duplicated_apps.append(name)
    else:
        unique_apps.append(name)

print('Number of duplicate apps: ', len(duplicated_apps))
print('\n')
print('Examples of duplicate apps: ', duplicated_apps[:15])

Number of duplicate apps:  1181


Examples of duplicate apps:  ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack', 'FreshBooks Classic', 'Insightly CRM', 'QuickBooks Accounting: Invoicing & Expenses', 'HipChat - Chat Built for Teams', 'Xero Accounting Software']


In total there is about 1181 duplicates.

However looking further in we can see that in the 4th column of the dataset we see that there is an increase in reviews so to speak. In this case we want keep one of the entires in the duplicate that has the highest number of reviews and then delete the rest of it. This can be used as a criteria for deleting rows of duplicates and keeping the most relavent one.

To do this we will:
- Use a dictonary to keep unique key app names, and the value with the highest number of reviews of that app.
- With the dictionary finished with the unique names with highest # of reviews we will then create a new data set, which will now contain one entry per app.

## Removing Duplicate Entries: Part Two

Starting off we will create a dictionary:

In [8]:
reviews_max = {}

for app in google_data:
    name = app[0]
    n_reviews = float(app[3])
    
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews
        
    elif name not in reviews_max:
        reviews_max[name] = n_reviews

In the previous example, we found that there are 1181 cases where there is a duplicate. So in this case our dictionary size should be the same as the difference between the lenght of the data set minus 1181.

In [9]:
print('Expected length:', len(google_data) - 1181)
print('Actual length:', len(reviews_max))

Expected length: 9659
Actual length: 9659


Now, we can do the following:
- Start by creating two empty lists:**android_clean** (which will store our new cleaned data set) and **already_added** (which will just store app names).
- Loop through the Google Play data set (make sure you don't include the header row), and for each iteration:
- Assign the app name to a variable named **name**.
- Convert the number of reviews to float, and assign it to a variable named **n_reviews**.
    - If **n_reviews** is the same as the number of maximum reviews of the app name (the number can be found in the reviews_max dictionary) and name is not already in the list **already_added** (read the solution notebook to find out why we need this supplementary condition):
        - Append the entire row to the **android_clean list** (which will eventually be a list of list and store our cleaned data set).
        - Append the name of the app name to the **already_added** list — this helps us to keep track of apps that we already added.

In [10]:
android_clean = []
already_added = []

for app in google_data:
    name = app[0]
    n_reviews = float(app[3])
    
    if (reviews_max[name] == n_reviews) and (name not in already_added):
        android_clean.append(app)
        already_added.append(name)

We can now explore the new data set, and confirm that the length is also 9569.

In [11]:
explore_data(android_clean, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9659
Number of columns: 13


And as expected we have now 9659 entries.

## Removing Non-English Apps

If we explore the dataset a bit more we notice that there are some non-english apps that suggest another audience. Below are some examples.

In [12]:
print(apple_data[813][1])
print(apple_data[6731][1])

print(android_clean[4412][0])
print(android_clean[7940][0])

爱奇艺PPS -《欢乐颂2》电视剧热播
【脱出ゲーム】絶対に最後までプレイしないで 〜謎解き＆ブロックパズル〜
中国語 AQリスニング
لعبة تقدر تربح DZ


For this analysis we are not interested in this demographic so these must be removed. One way of doing this is to remove apps whos names contain a symbol that is not commonly used in English Text.

In [13]:
def is_english(string):
    
    for character in string:
        if ord(character) > 127:
            return False
    
    return True

print(is_english('Instagram'))
print(is_english('爱奇艺PPS -《欢乐颂2》电视剧热播'))

True
False


This function works okay. However there are some apps that are in English that use special characters that are ouside the Ascii range of 127 like emoji and ™.

In [14]:
print(is_english('Docs To Go™ Free Office Suite'))
print(is_english('Instachat 😜'))

print(ord('™'))
print(ord('😜'))

False
False
8482
128540


## Removing Duplicate Entries: Part Two

To minimize the impact of data loss, we'll only remove apps that have more than 3 non-ASCII characters:

In [15]:
def is_english(string):
    non_ascii = 0
    
    for character in string:
        if ord(character) > 127:
            non_ascii += 1
    
    if non_ascii > 3:
        return False
    else:
        return True

print(is_english('Docs To Go™ Free Office Suite'))
print(is_english('Instachat 😜'))

True
True


This is still an okay solution but not perfect. However for now this seems a good enough filter for our analysis.

Below is the use of our newly edited is_english(string) function

In [16]:
android_english = []
ios_english = []

for app in android_clean:
    name = app[0]
    if is_english(name):
        android_english.append(app)
        
for app in apple_data:
    name = app[1]
    if is_english(name):
        ios_english.append(app)
        
explore_data(android_english, 0, 3, True)
print('\n')
explore_data(ios_english, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9614
Number of columns: 13


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 

We can see that there is 9614 Google Play and 6813 App Store entries left after the filtering.

## Isolating the Free Apps

Now we are going to be looking mroe closely whether or not the Apps are free to download or not. In this case for our study we want to look at the free ones.

In [17]:
android_final = []
ios_final = []

for app in android_english:
    price = app[7]
    if price == '0':
        android_final.append(app)
        
for app in ios_english:
    price = app[4]
    if price == '0.0':
        ios_final.append(app)
        
explore_data(android_final,0,1,True)
explore_data(ios_final,0,1,True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


Number of rows: 8864
Number of columns: 13
['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


Number of rows: 3222
Number of columns: 16


## Most Common Apps By Genre: Part One

To minimize risks and overhead, our validation strategy for an app idea is comprised of three steps:

- Build a minimal Android version of the app, and add it to Google Play.
- If the app has a good response from users, we develop it further.
- If the app is profitable after six months, we build an iOS version of the app and add it to the App Store.

Because our end goal is to add the app on both Google Play and the App Store, we need to find app profiles that are successful on both markets. For instance, a profile that works well for both markets might be a productivity app that makes use of gamification.

Let's begin the analysis by getting a sense of what are the most common genres for each market. For this, we'll need to build frequency tables for a few columns in our data sets.

## Most Common Apps By Genre: Part Two

We will build two functions we can use to analyze the frequency tables:

- One function to generate frequency tables that show percentages.
- Another function we can use to display the percenteges in a descending order.

In [18]:
def freq_table(dataset, index):
    table = {}
    
    for row in dataset:
        value = row[index]
        if value in table:
            table[value] += 1
        else:
            table[value] = 1
    
    table_percentages = {}
    for key in table:
        percentage = (table[key] /len(dataset)) * 100
        table_percentages[key] = percentage 
    return table_percentages


def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)
        
    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])

We have the function **display_table()**

- Takes in two parameters: **dataset** and **index**. **dataset** is expected to be a list of lists, and **index** is expected to be an integer.
- Generates a frequency table using the **freq_table()** function (which you're going to write as an exercise).
- Transforms the frequency table into a list of tuples, then sorts the list in a descending order.
- Prints the entries of the frequency table in descending order.

And the function **freq_table()**
- takes in two inputs: dataset (which is expected to be a list of lists) and index (which is expected to be an integer).
- The function should return the frequency table (as a dictionary) for any column we want. The frequencies should also be expressed as percentages.

## Most Common Apps by Genre: Part 3

In [19]:
display_table(ios_final, -5)

Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


Looking at the information we can see amoung the English apps, more than half (58.16%) are games. Entertainment is far from it but is second at (7.88%).

Making an inference on this information gives the impression that the App store is dominated by apps that are mostly for fun (as seen with the Games, Entertainment, Photo & video, social networking, etc), while those with pratical purposes (Utilities, education, sports, music, lifestyle, etc) are more on the rare side.

In this case we cannot be sure that even if the data set is populated with more apps for entertainment it doesnt mean that there is a great number of users - the demand for them might simply be overshadowed by the supply of them. (Might be too many games for the number of users play them on average)

In [20]:
display_table(android_final, 1)

FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

Looking on the android side of things we can see that it has a different side compare to the App Store. For one thing the catagory that populates the most is Family (18.9%). Making a further investigation we can see that the family catagory is mostly populated with kids games.

In [21]:
display_table(android_final, -4)

Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.591606498194946
Productivity : 3.892148014440433
Lifestyle : 3.892148014440433
Finance : 3.7003610108303246
Medical : 3.531137184115524
Sports : 3.463447653429603
Personalization : 3.3167870036101084
Communication : 3.2378158844765346
Action : 3.1024368231046933
Health & Fitness : 3.0798736462093865
Photography : 2.944494584837545
News & Magazines : 2.7978339350180503
Social : 2.6624548736462095
Travel & Local : 2.3240072202166067
Shopping : 2.2450361010830324
Books & Reference : 2.1435018050541514
Simulation : 2.0419675090252705
Dating : 1.861462093862816
Arcade : 1.8501805054151623
Video Players & Editors : 1.7712093862815883
Casual : 1.7599277978339352
Maps & Navigation : 1.3989169675090252
Food & Drink : 1.2409747292418771
Puzzle : 1.128158844765343
Racing : 0.9927797833935018
Role Playing : 0.9363718411552346
Libraries & Demo : 0.9363718411552346
Auto & Vehicles : 0.9250902527075

Looking in the Genre column of the data set we can see that at the top is Tools followed by entertainment, eduaction, business, production, lifestyle, etc.

Making a comparison of the patterns seen between the two data sets we can make an inference that on the google play store it is more for practical purposes based on the genre. However I also noticed that with the Google Play store that it has been divided into even more Genres compared to the App Store.

## Most Popular Apps by Genre on the App Store

To find out what genre are the most popular based on how many users have it, we need to calculate the average number of installs for each app genres. Thankfully with the Google Play data set, it is included in the **Installs** column, however the App Store data set is without this bit of information. 

For us, as a workaround we will be taking the total number of user ratings as a proxy, which can be found in the **ratings_count_tot**

The steps we will take is the following:

- Isolate the apps of each genre
- Sum up the users ratings for the apps of that genre.
- Divide the sum by the number of apps belonging to that genre



In [22]:
genre_ios = freq_table(ios_final, -5)

for genre in genre_ios:
    total = 0
    len_genre = 0
    for app in ios_final:
        genre_app = app[-5]
        if genre_app == genre:
            n_ratings = float(app[5])
            total += n_ratings
            len_genre += 1
    avg_n_ratings = total/len_genre
    print(genre, ':', avg_n_ratings)

Social Networking : 71548.34905660378
Photo & Video : 28441.54375
Games : 22788.6696905016
Music : 57326.530303030304
Reference : 74942.11111111111
Health & Fitness : 23298.015384615384
Weather : 52279.892857142855
Utilities : 18684.456790123455
Travel : 28243.8
Shopping : 26919.690476190477
News : 21248.023255813954
Navigation : 86090.33333333333
Lifestyle : 16485.764705882353
Entertainment : 14029.830708661417
Food & Drink : 33333.92307692308
Sports : 23008.898550724636
Book : 39758.5
Finance : 31467.944444444445
Education : 7003.983050847458
Productivity : 21028.410714285714
Business : 7491.117647058823
Catalogs : 4004.0
Medical : 612.0


On average we can see that Reference is a good genre profile to look at with this data set. We can make a closer inspection by looking at a few app in this genre.

In [23]:
for app in ios_final:
    if app[-5] == "Reference":
        print(app[1], ":", app[5])

Bible : 985920
Dictionary.com Dictionary & Thesaurus : 200047
Dictionary.com Dictionary & Thesaurus for iPad : 54175
Google Translate : 26786
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588
Merriam-Webster Dictionary : 16849
Night Sky : 12122
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693
GUNS MODS for Minecraft PC Edition - Mods Tools : 1497
Guides for Pokémon GO - Pokemon GO News and Cheats : 826
WWDC : 762
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 718
VPN Express : 14
Real Bike Traffic Rider Virtual Reality Glasses : 8
教えて!goo : 0
Jishokun-Japanese English Dictionary & Translator : 0


As we can see with this category a good portion of these downloads can be seen with apps that refer to the bible with a close second being anything dealing with the dictionary. However as we can see these are very popular apps that people use on a day to day basis sometimes. We want to find a popular app that doesn't have a big of an impact and that doesn't have this big skewed compared to the other ones.

There are other genres that seem popular. Some of theres are: Social Networking, Music, Food & Drinks, Navigation, Weather. These fit in with the the suggestion that the App Store mostly have apps that are in the category of fun with the exception of Navigation and Weather. 

Weather - people just generally use this to check the weather at off times if they remember too. So this does not seem like a good place to make profit.

Food & Drink - This might have some problems due to popular apps like Uber-Eats, SkipTheDishes, PostMates, and other food delivery services that introducing a new app in this genre will be very hard might come as a loss if we have no influence.

We now want to move on to the Google Play market.

## Most Popular Apps by Genre on Google Play

For the Google Play markets luckily we have a column that includes the amount of installs an app has so it cuts down our work to just analyzing.

In [24]:
display_table(android_final, 5)

1,000,000+ : 15.726534296028879
100,000+ : 11.552346570397113
10,000,000+ : 10.548285198555957
10,000+ : 10.198555956678701
1,000+ : 8.393501805054152
100+ : 6.915613718411552
5,000,000+ : 6.825361010830325
500,000+ : 5.561823104693141
50,000+ : 4.7721119133574
5,000+ : 4.512635379061372
10+ : 3.5424187725631766
500+ : 3.2490974729241873
50,000,000+ : 2.3014440433213
100,000,000+ : 2.1322202166064983
50+ : 1.917870036101083
5+ : 0.78971119133574
1+ : 0.5076714801444043
500,000,000+ : 0.2707581227436823
1,000,000,000+ : 0.22563176895306858
0+ : 0.04512635379061372
0 : 0.01128158844765343


Even with it having a install column it is hard to be precise with this type of data. For one thing there is no indication on which apps are at 1,000,000+ installs and so on. However we do not need to be absolutely precise for this analysis so we can work with this data.

In [29]:
category_android = freq_table(android_final, 1)

for category in category_android:
    total = 0
    len_category = 0
    for app in android_final:
        if app[1] == category:
            n_install = app[5]
            n_install = n_install.replace('+', '')
            n_install = n_install.replace(',', '')
            total += float(n_install)
            len_category += 1
    avg_n_installs = total / len_category
    print(category, ':', avg_n_installs)

ART_AND_DESIGN : 1986335.0877192982
AUTO_AND_VEHICLES : 647317.8170731707
BEAUTY : 513151.88679245283
BOOKS_AND_REFERENCE : 8767811.894736841
BUSINESS : 1712290.1474201474
COMICS : 817657.2727272727
COMMUNICATION : 38456119.167247385
DATING : 854028.8303030303
EDUCATION : 1833495.145631068
ENTERTAINMENT : 11640705.88235294
EVENTS : 253542.22222222222
FINANCE : 1387692.475609756
FOOD_AND_DRINK : 1924897.7363636363
HEALTH_AND_FITNESS : 4188821.9853479853
HOUSE_AND_HOME : 1331540.5616438356
LIBRARIES_AND_DEMO : 638503.734939759
LIFESTYLE : 1437816.2687861272
GAME : 15588015.603248259
FAMILY : 3695641.8198090694
MEDICAL : 120550.61980830671
SOCIAL : 23253652.127118643
SHOPPING : 7036877.311557789
PHOTOGRAPHY : 17840110.40229885
SPORTS : 3638640.1428571427
TRAVEL_AND_LOCAL : 13984077.710144928
TOOLS : 10801391.298666667
PERSONALIZATION : 5201482.6122448975
PRODUCTIVITY : 16787331.344927534
PARENTING : 542603.6206896552
WEATHER : 5074486.197183099
VIDEO_PLAYERS : 24727872.452830188
NEWS_AND_

Looking at the average, communication apps have the most installs at: 38456119. This number is skewed with apps like WhatsApp, Facebook Messenger, Skype, Gmail, Hangout, Discord, and with the inclusion of Zoom for students), and the other apps cover very small percentages.

In [30]:
for app in android_final:
    if app[1] == 'COMMUNICATION' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

WhatsApp Messenger : 1,000,000,000+
imo beta free calls and text : 100,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Skype - free IM & video calls : 1,000,000,000+
Who : 100,000,000+
GO SMS Pro - Messenger, Free Themes, Emoji : 100,000,000+
LINE: Free Calls & Messages : 500,000,000+
Google Chrome: Fast & Secure : 1,000,000,000+
Firefox Browser fast & private : 100,000,000+
UC Browser - Fast Download Private & Secure : 500,000,000+
Gmail : 1,000,000,000+
Hangouts : 1,000,000,000+
Messenger Lite: Free Calls & Messages : 100,000,000+
Kik : 100,000,000+
KakaoTalk: Free Calls & Text : 100,000,000+
Opera Mini - fast web browser : 100,000,000+
Opera Browser: Fast and Secure : 100,000,000+
Telegram : 100,000,000+
Truecaller: Caller ID, SMS spam blocking & Dialer : 100,000,000+
UC Browser Mini -Tiny Fast Private & Secure : 100,000,000+
Viber Mess

If we do omit the apps that generate over 100 million installs the average will be reduced by quite a bit.

In [36]:
under_100_m = []

for app in android_final:
    n_installs = app[5]
    n_installs = n_installs.replace(',', '')
    n_installs = n_installs.replace('+', '')
    if (app[1] == 'COMMUNICATION') and (float(n_installs) < 100000000):
        under_100_m.append(float(n_installs))
        
sum(under_100_m) / len(under_100_m)

3603485.3884615386

In [39]:
38456119 // 3603485

10

As we can see, our estimation was correct where if we removed those apps that generate more than 100,000,000 installs it is around a 10 times difference compare to if we didn't remove it. This will be a slightly hard to build an app for this market as there are already giants that have been estabilshed. But not to say that it will be hard because if it were to be successful a huge number of people could convert to using this app. For example with the introduction of Discord I believe there has been a huge reduction in people using Skype and with Zoom also being introduced into the mix it made it even harder for Skype to come back.

These pattern can also be seen with other apps due to some having a huge possible market due to there number of installs. This can also influence what is seen at the front page of the Google Play store. If an app gets popular then it has a chance to get to the front page which helps it grow as people pass by it more often then other apps.

Another example we can see games:

In [44]:
for app in android_final:
    n_installs = app[5]
    n_installs = n_installs.replace('+','')
    n_installs = n_installs.replace(',','')
    if app[1] == 'GAME' and float(n_installs) > 100000000:
        print(app[0] ,':', app[5])

Candy Crush Saga : 500,000,000+
Subway Surfers : 1,000,000,000+
Pou : 500,000,000+
My Talking Tom : 500,000,000+
Temple Run 2 : 500,000,000+


For games themselves if we look at the most popular ones they seem to be the old classics that have been on the Google Play store for some time. This might be a good market to head into but it is also flooded with so many games that people might pass over it while searching.

In [45]:
for app in android_final:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000+'
                                            or app[5] == '5,000,000+'
                                            or app[5] == '10,000,000+'
                                            or app[5] == '50,000,000+'):
        print(app[0], ':', app[5])

Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
AlReader -any text book reader : 5,000,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
ReadEra – free ebook reader : 1,000,000+
Ebook Reader : 5,000,000+
Read books online : 5,000,000+
eBoox: book reader fb2 epub zip : 1,000,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
Moon+ Reader : 10,000,000+
English-Myanmar Dictionary : 1,000,000+
Golden Dictionary (EN-AR) : 1,000,000+
All Language Translator Free : 1,000,000+
Aldiko Book Reader : 10,000,000+
Dictionary - WordWeb : 5,000,000+
50000 Free eBooks & Free AudioBooks : 5,000,000+
Al-Quran (Free) : 10,000,000+
Al Quran Indonesia : 10,000,000+
Al'Quran Bahasa Indonesia : 10,000,000+
Al Quran Al karim : 1,000,000+
Al Quran : EAlim - Translations & MP3 Offline : 5,000,000+
Koran Read &MP3 30 Juz Offline : 1,000,000+
H

Looking over at what is avalible in the Books and Reference catagory we can see that it is mostly dominate with apps that have a large collection of ebooks and ebook readers as well as dictonaries for on the go access. This is not a good market to look at as there is going to be competition with other estabished apps.

One good thing to note is that they have apps around the book of Quran. This seems like a market to get into where taking a popuilar book and turning it into an app could be profitable.

However we need to add something special because there is already various numbers of digital libraries which will defeat the purpose of making an app solely on just raw books translated into an app.

## Conclusion

This project was used to analyze market data in both Google Play and App Store with the goal of recommending an app profile for both markets.

We can see that it is a good chance that turning well known books into apps can make a good profit in both markets however we need to make it more with features due to the fact that there is already estabished digital libraries that defeats the purpose of creating an app solely on just a book itself.